sklearn/utils/estimator_checks.py

```diff
@@ -1019,6 +1034,7 @@ def check_array_api_input_and_values(
     array_namespace,
     device=None,
     dtype_name="float64",
+    skip_methods={},
```

Comment:
I think we should use `skip_methods=None` + `if skip_methods is None: skip_methods = {}` in the function body. Mutable values as default value for a function argument is a trap waiting for someone to fall into.

<details>
<summary>Details</summary>

```python
def foo(x, a=[]):
  a.append(x)
  print(a)

foo(1)  # prints [1]
foo(1)  # print [1, 1]
```

</details>